lib/ansible/module_utils/basic.py

```diff
@@ -1849,20 +1851,10 @@ def _handle_options(self, argument_spec=None, params=None, prefix=''):
                 self._options_context.pop()
 
     def _get_wanted_type(self, wanted, k):
-        if not callable(wanted):
-            if wanted is None:
-                # Mostly we want to default to str.
-                # For values set to None explicitly, return None instead as
-                # that allows a user to unset a parameter
-                wanted = 'str'
-            try:
-                type_checker = self._CHECK_ARGUMENT_TYPES_DISPATCHER[wanted]
-            except KeyError:
-                self.fail_json(msg="implementation error: unknown type %s requested for %s" % (wanted, k))
-        else:
-            # set the type_checker to the callable, and reset wanted to the callable's name (or type if it doesn't have one, ala MagicMock)
-            type_checker = wanted
-            wanted = getattr(wanted, '__name__', to_native(type(wanted)))
+        try:
+            type_checker = get_wanted_type(wanted)
```

Comment:
I think this muddies your API design where the `check_*` methods raise exceptions on error and the `get_*` methods are not exception throwers and always return data instead. So you could either rename it to `check_wanted_type()` or have it return None or empty string for unknown types.

Reply:
That makes sense. I'll make the change.